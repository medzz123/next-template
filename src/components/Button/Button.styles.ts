import { styled } from '@theme/config';

export const ButtonContainer = styled('button', {
  backgroundColor: '$purple600',
});
