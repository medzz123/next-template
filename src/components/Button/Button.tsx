import React, { FunctionComponent } from 'react';

import { ButtonProps } from './Button.models';
import { ButtonContainer } from './Button.styles';

const Button: FunctionComponent<ButtonProps> = (props): React.ReactElement => {
  const { children } = props;

  return <ButtonContainer data-testid="button">{children}</ButtonContainer>;
};

export default Button;
