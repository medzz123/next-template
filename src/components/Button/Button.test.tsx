import { render } from '@test/testUtils';
import React from 'react';

import Button from './Button';

test('Should render button', () => {
  const { container } = render(<Button>Hello</Button>);

  expect(container).toMatchSnapshot();
});
