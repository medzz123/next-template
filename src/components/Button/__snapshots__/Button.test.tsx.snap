// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test 1`] = `
<div>
  <button
    class="sxf40k1"
    data-testid="button"
  >
    Hello
  </button>
</div>
`;
